tests/test_cli.py

```diff
@@ -0,0 +1,244 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+
+"""CLI tests."""
```

Comment:
(One general CLI API testing comment that may lead to changing the Pythonic API implementation.)

When asking to dereference a non-existing PID, the CLI always raises `PIDDoesNotExistError` both when the PID does not exist and when the PID type is wrong, for example:

```console
$ invenio3 pid object recid 12
rec 5dc39e3c-b300-11e5-8db9-08002710a880 R

$ invenio3 pid object recid 123456789
Traceback (most recent call last):
[...]
invenio_pidstore.errors.PIDDoesNotExistError

$ invenio3 pid object recidfoo 12
Traceback (most recent call last):
[...]
invenio_pidstore.errors.PIDDoesNotExistError
```

It would be advantageous to return two different error statuses in this case, e.g. 1 for unknown PID type and 2 for unknown PID value.